doas.c

```diff
@@ -183,25 +183,35 @@ parseconfig(const char *filename, int checkperms)
 }
 
 #ifdef DOAS_CONFDIR
+static int
+filterconfdir(const struct dirent *dirent)
+{
+	return dirent->d_type == DT_REG;
```

Comment:
I think this is relevant:

```
 Currently, only some filesystems (among them: Btrfs, ext2,
 ext3, and ext4) have full support for returning the file type
 in d_type.  All applications must properly handle a return of
 DT_UNKNOWN.
```

And someone doing system configuration via something like `stow` could have a symlink here.

Not sure it makes sense to filter at all? Or use `lstat`, though it's more expensive.

Reply:
Arguably checking if names end in `.conf` is even more important.

(I think we need a man page change to discuss the UX aspect fully)

Reply:
I don't think requiring names to end in `.conf` is important.  We presently don't have that requirement with `/etc/sudoers.d`.

I thought `DT_REG` was required by POSIX to work, but it appears `d_type` is actually *not* part of POSIX at all.  I'll rework it to use lstat(2).

Reply:
Also, if we don't filter, we will get things that parseconfig() will choke on.  If you hand it a directory, it will fail, and rightly exit.

Reply:
I think there should be a suffix check otherwise we run into problems with package managers that consider those files as configuration file and extract changed defaults as `.new-$version` etc.

Reply:
Fair enough.  One moment...